tools/ras/main.c

```diff
@@ -1067,7 +1067,7 @@ fpga_result mmio_error(struct RASCommandLine *rasCmdLine)
 	if ( rasCmdLine->function >0 )
 		function = rasCmdLine->bus;
 
-	snprintf(sysfs_path, sizeof(sysfs_path),
+	snprintf_s_iiii(sysfs_path, sizeof(sysfs_path),
```

Comment:
Can you explain why is this necessary? Is `snprintf()` with four integer arguments unsafe?

Reply:
I see.

In this case the format string is a constant containing four `%i`, and the arguments are either constants (`0`) or `int` - is `sprintf_iiii()` necessary in this case?

Reply:
I think so, or else we could try to get a per-instance waiver; even then, we would still have to deal with the issue when Klocwork flags it.